js/coinsbit.js

```diff
@@ -0,0 +1,1073 @@
+        //          success: true,
+        //          message: "",
+        //          result: {
+        //            bid: '46504.7412435',
+        //            ask: '46511.54203562',
+        //            open: '47030.33',
+        //            high: '47549.99',
+        //            low: '45571.08',
+        //            last: '46459.99',
+        //            volume: '8372.29022355',
+        //            deal: '388090406.73607745',
+        //          },
+        //          code: "200"
+        //     }
+        ticker = this.safeValue (response, 'result', {});
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetTickers (params);
+        //     {
+        //          success: true,
+        //          message: "",
+        //          result: {
+        //            BTC_USDT: {
+        //               at: '1640028811',
+        //               ticker: {
+        //                 bid: '46448.49682539',
+        //                 ask: '46458.41093569',
+        //                 open: '46687.39',
+        //                 high: '47550',
+        //                 low: '45584.27',
+        //                 last: '46462.77',
+        //                 vol: '32.56275846',
+        //                 deal: '1510574.62551143',
+        //                 change: '-0'
+        //               }
+        //            },
+        //            ...
+        //          },
+        //          code: "200"
+        //     }
+        const tickers = this.safeValue (response, 'result', {});
+        const result = [];
+        const marketIds = Object.keys (tickers);
+        for (let i = 0; i < marketIds.length; i++) {
+            const marketId = marketIds[i];
+            const market = this.safeMarket (marketId);
+            result.push (this.parseTicker (tickers[marketId], market));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //       at: '1640028811',
+        //       ticker: {
+        //         bid: '46448.49682539',
+        //         ask: '46458.41093569',
+        //         open: '46687.39',
+        //         high: '47550',
+        //         low: '45584.27',
+        //         last: '46462.77',
+        //         vol: '32.56275846',
+        //         deal: '1510574.62551143',
+        //         change: '-0'
+        //       }
+        let tickerFinal = {};
+        let timestamp = undefined;
+        if ('ticker' in ticker) { // if arrived from fetchTickers
+            tickerFinal = this.safeValue (ticker, 'ticker', {});
+            timestamp = this.safeTimestamp (ticker, 'at');
+        } else { // if arrived from fetchTicker, then there is nothing to be done more
+            tickerFinal = ticker;
+            timestamp = this.milliseconds ();
```

Comment:
we cannot set timestamp to `this.milliseconds` in the library, since we don't really want to make up data that the exchange doesn't provide

Reply:
@frosty00 
ah, right. I'll push the changes.
However,  if you remember, in other thread we had a discussion - what about creating additional property (i.e. 'emulatedTS' which will be filled with emulated timestamp when exchange TS is not available?  I think _in the end of day_ users will need some approximate value for their generic (for not rocket-science algos) analysis etc, so if that emulated TS will be filled with something like exchange response header's TS (or the middle point of the roundtrip)?

Reply:
Yep, we will think of adding an optional property for that, but it's better if we do that outside of this specific Coinsbit PR.

Reply:
@kroitor  sure, I've meant that actually, not in this specific PR, this is not right place right time.